src/StoreApi/Routes/V1/CartUpdateCustomer.php

```diff
@@ -215,16 +216,23 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 	 * @return array
 	 */
 	protected function get_customer_billing_address( \WC_Customer $customer ) {
+		$validation_util = new ValidationUtils();
+		$billing_country = $customer->get_billing_country();
+		$billing_state   = $customer->get_billing_state();
+
+		if ( ! $validation_util->validate_state( $billing_state, $billing_country ) ) {
```

Comment:
We need an inline comment here explaining the need for this fix/why core breaks things. Otherwise we'll definitely forget.